packages/calypso-e2e/src/lib/pages/plans-page.ts

```diff
@@ -0,0 +1,85 @@
+import { Page } from 'playwright';
+import { getViewportName } from '../../browser-helper';
+
+export type Plan = 'Free' | 'Personal' | 'Premium' | 'Business' | 'eCommerce';
+export type PlansPageTab = 'My Plan' | 'Plans';
+export type PlanActionButton = 'Manage plan' | 'Upgrade';
+
+const selectors = {
+	myPlanTitle: '.my-plan-card__title',
+	navigationTab: '.section-nav-tab',
+	activeNavigationTab: '.is-selected.section-nav-tab',
+	mobileActionButton: '.plan-features__mobile >> .plan-features__actions-button',
+	desktopActionButtion: '.plan-features__table >> .plan-features__actions-button',
+};
+
+/**
+ * Page representing the Plans page accessible at Upgrades >> Plans
+ */
+export class PlansPage {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the Plans POM.
+	 *
+	 * @param {Page} page Instance of the Playwright page
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Validates that the provided plan name is the title of the active plan in the My Plan tab of the Plans page. Throws if it isn't.
+	 *
+	 * @param expectedPlan Name of the expected plan.
+	 * @throws If the expected plan title is not found in the timeout period.
+	 */
+	async validatePlanInMyPlan( expectedPlan: Plan ): Promise< void > {
+		await this.page.waitForSelector( `${ selectors.myPlanTitle }:has-text("${ expectedPlan }")` );
+	}
+
+	/**
+	 * Validates that the provided tab name is the the currently active tab in the wrapper Plans page. Throws if it isn't.
+	 *
+	 * @param expectedTab Name of the expected tab.
+	 * @throws If the expected tab name is not the active tab.
+	 */
+	async validateActiveNavigationTab( expectedTab: PlansPageTab ): Promise< void > {
+		await this.page.waitForSelector(
+			`${ selectors.activeNavigationTab }:has-text("${ expectedTab }")`
+		);
+	}
```

Comment:
I can't decide if I want to do these this way, where we do a text-based selector, and let `waitForSelector` throw, or alternatively grab and return the actual content of the text nodes at these locations, then have the tests themselves assert (using Jest `expect`) against that string to make sure it is the expected value.